bowtie/_cli.py

```diff
@@ -1271,6 +1277,179 @@ def validate(
     return asyncio.run(_run(fail_fast=False, **kwargs, cases=[case]))
 
 
+@subcommand
+@IMPLEMENTATION
+@dialect_option()
+@format_option()
+@click.option(
+    "--runs",
+    "-r",
+    "runs",
+    type=int,
+    default=3,
+    show_default=True,
+    help="Number of runs used to run benchmarks.",
+)
+@click.option(
+    "--values",
+    "-v",
+    "values",
+    type=click.IntRange(min=2),
+    default=2,
+    show_default=True,
+    help="Number of values per run.",
+)
+@click.option(
+    "--warmups",
+    "-w",
+    "warmups",
+    type=int,
+    default=1,
+    show_default=True,
+    help="Number of skipped values per run used to warmup the benchmark.",
+)
+@click.option(
+    "--loops",
+    "-l",
+    "loops",
+    type=int,
+    default=1,
+    show_default=True,
+    help="Number of loops per value.",
+)
+@click.option(
+    "--quiet",
+    "-q",
+    "quiet",
+    is_flag=True,
+    default=False,
+    show_default=True,
+    help="Enable quiet mode (Only output the final result).",
+)
+@click.option(
+    "--keywords",
+    "-k",
+    "keywords",
+    is_flag=True,
+    default=False,
+    show_default=True,
+    help=(
+        "Run keyword specific benchmarks to learn about how "
+        "various implementations implement the keyword."
+    ),
+)
+@click.option(
+    "-b",
+    "--benchmark-file",
+    "benchmark_files",
+    multiple=True,
+    help=(
+        "Allows running benchmark from a file. "
+        "Specify the path of the benchmark file"
+    ),
+)
+@click.option(
+    "--test-suite",
+    "-t",
+    "test_suite",
+    type=_suite.ClickParam(),
+    default=None,
+    help="Run Benchmarks over the official JSON Schema Test Suite.",
+)
+@click.argument("benchmark", type=JSON(), required=False)
+def perf(
+    connectables: Iterable[_connectables.Connectable],
+    dialect: Dialect,
+    format: _F,
+    keywords: bool,
+    benchmark: dict[str, Any] | None,
+    quiet: bool,
+    test_suite: tuple[Iterable[TestCase], Dialect, dict[str, Any]],
+    benchmark_files: Iterable[str],
+    **kwargs: Any,
+):
+    """
+    Perform performance measurements across supported implementations.
+    """
+    try:
+        if benchmark_files:
```

Comment:
Generally I think it's nicer in general to avoid `if/elif` chains -- specifically if you want a small exercise, here I think the pattern would be similar to what we do elsewhere, namely when you define each of these options, they shouldn't set boolean flags, they should set a `deserialize` argument (so `keywords` sets it to `Benchmarker.for_keywords` or some wrapper callable for that, `test_suite` to `from_test_cases`, etc.)

Reply:
Hi @Julian , tried doing this. Can you please have a look ?

Reply:
The arguements for each are different but, any idea how to handle that?

Reply:
Could you please have a look now? @Julian